### ISSUE TITLE
Installation problems on Ubuntu

### ISSUE THREAD

**Issue by lukecampbell** [ioos/compliance-checker#65]
There is an issue installing compliance checker on Ubuntu (Version to be filled in).

Installing udunitspy, a dependency of compliance checker, is reporting the lack of a swig executable.

SWIG is a library or tool that is used to create bindings, or an interface between a python module and a dynamically linked library written in C/C++ (or another compiled language) that is usually native to the operating system.

The issue was originally reported by @mbiddle-nodc

**Comment by abirger** [comment]
@lukecampbell , tried the Ubuntu instruction with the Linux Mint 32bit running in VirtualBox. The udunits installation failed at './config':

```
oos@ioos-VirtualBox ~/udunits-2.1.24 $ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking for C compiler default output file name... 
configure: error: in `/home/ioos/udunits-2.1.24':
configure: error: C compiler cannot create executables
See `config.log' for more details.
```

**Comment by abirger** [comment]
@lukecampbell , @kknee , @rsignell-usgs , @dpsnowden , @carmelortiz , here is the summary of my compliance-checker installation with the Linux in VirtualBox (the mileage may vary in case of native Linux):
With 32bit Linux Mint - complete failure:
- "conda" method did not work with 32bit Linux because only rsignell/udunitspy for 64bit exists.
- "pip" method stopped when udunits compilation failed:
```
ioos@ioos-VirtualBox ~/udunits-2.1.24 $ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking for C compiler default output file name... 
configure: error: in `/home/ioos/udunits-2.1.24':
configure: error: C compiler cannot create executables
```
With 64bit Linux Mint - partial success:
- "conda" method worked; however, there is a confusion between 'compliance-checker' and 'compliance_checker'. The commands from README.md
```
conda create --file requirements.txt -n compliance-checker    
conda install compliance-checker    
source activate compliance_checker
```
resulted in error "could not find environment: compliance_checker".

On the other hand, the following worked:
```
conda create --file requirements.txt -n compliance_checker    
conda install compliance-checker    
source activate compliance_checker
```
In any case, I had to run `conda install compliance-checker` again after virtual environment activation; otherwise the test commands from README.md failed with error "compliance-checker: command not found". Even after I re-install compliance-checker from the virtualenv, I had to provide the absolute path to the .nc files to make the other 2 test commands work.

All in all, the installation is not smooth, and some steps are **REALLY** confusing and time-consuming for people like me who are not Linux-oriented...

**Comment by dpsnowden** [comment]
I view this as further evidence that as we develop software, we need  a little more attention to dependencies, documentation, cross platform support, etc.  Our goal should be to test these types of packages in environments average science users use and not professional software engineers.  I'm  not sure there is a direct action other than incorporate this level of testing and user support in our ongoing project plans.

**Comment by abirger** [comment]
@daf , thanks a lot! Will it fix the path issue for the provided test command
```
compliance-checker --test=acdd compliance_checker/tests/data/ru07-20130824T170228_rt0.nc
```
or should I normally use the full path to the file? Actually, I tried appending the PATH variable but regardless of what I put there the command failed with relative path suggested...

**Comment by daf** [comment]
@abirger, I think the README is assuming you were in the compliance-checker's development directory, and is using a relative path from there.  You'll need to specify a full path, as $PATH only deals with resolving where executables are.  I can't remember if the compliance-checker package will install the test .nc files too, it probably does.

**Comment by abirger** [comment]
@daf , the README test commands work as-is only if the working directory is `~/anaconda/envs/compliance_checker/lib/python2.7/site-packages`. Apparently, compliance-checker installs test files, and other stuff into `~/anaconda/envs/compliance_checker/lib/python2.7/site-packages/compliance_checker`, which seems strange.

**Comment by rsignell-usgs** [comment]
@abirger, that is normal for anaconda.   Each custom environment has has it's own directory that contains all the packages that environment has in `<anaconda>/envs/<custom_env_name>/lib/site-packages`

**Comment by daf** [comment]
The instructions should also be amended to not reference files as if you're working in the development directory.  Should probably mention instructions to download one of the test nc files from github and run it on that.  Or just test via OPeNDAP.

**Comment by daf** [comment]
udunitspy/SWIG is no longer a requirement, closing this issue.  Please file a new, specific issue for further issues.
